lib/Array.fx

```diff
@@ -61,6 +61,18 @@ operator .% (a: 'ta [+], b: 'tb [+]) =
 operator .** (a: 'ta [+], b: 'tb [+]) =
     [| for x <- a, y <- b {x .** y} |]
 
+operator += (a: 'ta [+], b: 'tb) { for x@idx <- a { a[idx] = x + b} }
```

Comment:
looks great! At some point it would be nice to add some optimization into `K_fast_idx` and `C_gen_code` to:

1. remove range checks when `for ...@idx <- arr { arr[idx] = ... }` is used (regardless of whether `idx` is 1D or n-D)
2. use temporary pointer and fast `ptr[loop_var]` construction instead of heavy n-d array element access macro when `for ... { arr[i0, i1, ..., i(n-1)] = ... }` is used and `i0, ..., i(n-2)` are invariants of the most-nested loop, e.g.

Of course, that should be a different pull request and we need to put a proper feature request into the bug tracker